main.rb

```diff
@@ -0,0 +1,130 @@
+# rubocop:disable Metrics/CyclomaticComplexity,Metrics/PerceivedComplexity
+
+module Enumerable
+  def my_each
+    return to_enum(__method__) unless block_given?
+
+    my_iterable = enforce_arr
+    arr_len = my_iterable.length
+    arr_len.times do |index|
+      yield(my_iterable[index])
+    end
+    my_iterable
+  end
+
+  def my_each_with_index
+    return to_enum(__method__) unless block_given?
+
+    my_iterable = enforce_arr
+    arr_len = my_iterable.length
+    arr_len.times do |index|
+      yield(my_iterable[index], index)
+    end
+    my_iterable
+  end
```

Comment:
- [ ]  When a block is given and `my_each` is called on a range, it should return the range itself after calling the given block once for each element in the range. 
    - Like your `my_each` yours is returning an array of numbers rather than a range. 

- [ ] When `my_each_with_index` is called on a hash and a block is given, it should call the block once for each element in the hash